dace/transformation/helpers.py

```diff
@@ -687,6 +687,85 @@ def state_fission(sdfg: SDFG, subgraph: graph.SubgraphView, label: Optional[str]
     return newstate
 
 
+def state_fission_after(sdfg: SDFG, state: SDFGState, node: nodes.Node, label: Optional[str] = None) -> SDFGState:
+    """
+    """
+    newstate = sdfg.add_state_after(state, label=label)
+
+    # Bookkeeping
+    successors = set([node])
+    boundary_nodes = set()
+    orig_edges = set()
+
+    # Collect predecessors
+    if not isinstance(node, nodes.AccessNode):
+        for edge in state.in_edges(node):
+            for e in state.memlet_path(edge):
+                successors.add(e.src)
```

Comment:
This does not look like a successor. Perhaps what you want is to call it something like nodes_to_keep (for new state)